tests/rptest/tests/delete_records_test.py

```diff
@@ -772,10 +772,10 @@ class PartitionMoveExceptionReporter:
         partition_move_thread.join()
         self.redpanda.logger.info("Joining on producer thread")
         producer.wait()
-        self.redpanda.logger.info("Calling consumer::stop")
-        consumer.stop()
         self.redpanda.logger.info("Joining on consumer thread")
         consumer.wait()
+        self.redpanda.logger.info("Calling consumer::stop")
+        consumer.stop()
```

Comment:
Done. Also added another commit.

In most of the services and judging by ducktape code doc and the way stop() is used it is really intended to hard stop a service. It is also always called before start() calls to kill potentially previous instances of the service lingering around.

However, wait()

```
Wait for the service to finish.
This only makes sense for tasks with a fixed amount of work to do. For services that generate
output, it is only guaranteed to be available after this call returns.
```

So in this case, since we really want the consumer to finish its work for asserts we make about invalid reads count to make sense. Stopping, in the general case, does not even guarantee it started doing any work...

And kgo is implemented exactly this way. Stopping kills the service and resets it's status thread. Waiting just short circuits if the status thread is not available anymore since it knows the service was already stopped. In a later commit in this PR I make this sequence of calls invalid to prevent future misuse of the service.